src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs

```diff
@@ -241,12 +241,16 @@ private void UpdateCore()
                 renderTransform *= (-offset) * visual.RenderTransform.Value * (offset);
             }
 
-
-
             comp.TransformMatrix = MatrixUtils.ToMatrix4x4(renderTransform);
 
             _recorder.BeginUpdate(comp.DrawList);
+
+            _recorder.PushRenderOptions(visual._renderOptions);
```

Comment:
This adds 2 extra drawing nodes for every single visual, even to ones that don't have any content at all. That will cause severe memory traffic on rendering.

Reply:
Y by removing push / pop we eliminate that